src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -205,37 +203,35 @@ export function DiscoverMainRoute(props: Props) {
     );
   }, [chrome, savedSearch]);
 
-  if (showNoDataPage) {
-    const analyticsServices = {
-      coreStart: core,
-      dataViews: {
-        ...data.dataViews,
-        hasData: {
-          ...data.dataViews.hasData,
-
-          // We've already called this, so we can optimize the analytics services to
-          // use the already-retrieved data to avoid a double-call.
-          hasESData: () => Promise.resolve(isDev ? true : hasESData),
-          hasUserDataView: () => Promise.resolve(hasUserDataView),
-        },
+  const analyticsServices = {
+    coreStart: core,
+    dataViews: {
+      ...data.dataViews,
+      hasData: {
+        ...data.dataViews.hasData,
+        // If we're in dev mode, we want to always return true here.
+        hasESData: isDev ? () => Promise.resolve(true) : data.dataViews.hasData.hasESData,
```

Comment:
I could reproduce the CI failure by setting this to 

```suggestion
        hasESData: data.dataViews.hasData.hasESData,
```
If you then create a new data view when "Now, create a data view" is displayed in Discover, this screen doesn't go away, when you did create one. You've to do a browser refresh.

This is because of  `isDev === true` . in this case every time `analyticsServices` is assigned, hasESData is assigned a new function, while when `isDev === false` `data.dataViews.hasData.hasESData` is assigned, which is always the same function. 

As a result the state of KibanaNoDataPage is not updated, because `useEffect` is not triggered, since `[hasESData, hasUserDataView]` doesn't change:

https://github.com/elastic/kibana/blob/a1fd3307b94f388e3e5fa9a05a4f52bc1ad19f36/packages/shared-ux/page/kibana_no_data/impl/src/kibana_no_data_page.tsx#L32-L45

The solution could be to:

```suggestion
        hasESData: () => (isDev ? Promise.resolve(true) : data.dataViews.hasData.hasESData),
```
then the useEffect would work as expected